src/coreclr/jit/compiler.h

```diff
@@ -6508,7 +6511,7 @@ class Compiler
         {
             if (lpHead->NextIs(lpEntry))
             {
-                assert(lpHead->bbFallsThrough());
+                assert(lpHead->bbFallsThrough() || lpHead->JumpsToNext());
                 assert(lpTop == lpEntry);
                 return true;
```

Comment:
Unrelated to your PR, but it seems odd to me that we couldn't have a bottom-entered loop where `lpHead` immediately precedes the top block.